src/validation.h

```diff
@@ -191,7 +211,7 @@ struct PackageMempoolAcceptResult
 {
     const PackageValidationState m_state;
     /**
-    * Map from wtxid to finished MempoolAcceptResults. The client is responsible
+    * Map from (w)txid to finished MempoolAcceptResults. The client is responsible
```

Comment:
I think this change can be problematic. Now we don't know what the keys are for this map (txid or wtxid).
Are callers expected to just try both?

Reply:
Essentially yes, so that we can handle same-txid-different-witness. It's "if you can't find it by wtxid then try txid because it may have been swapped out."

I also considered adding a special result type like `ResultType::WITNESS_SWAPPED` to communicate  "try again using txid" to the caller (would that be better?).

Another approach is to include the entire transaction in the `MempoolAcceptResult`, but it seems weird if the result transaction's wtxid doesn't match the key wtxid.

It's a little ugly, but I'm not sure if there's a cleaner way to do this?

Reply:
> I also considered adding a special result type like ResultType::WITNESS_SWAPPED to communicate "try again using txid" to the caller (would that be better?).

It does feel better to me, but there's a part of your answer I don't understand.
This case only happens when the mempool contain a "same txid, different wtxid" entry, right?
Let's call A our proposed tx and A' the mempool "same txid, different wtxid" tx.
Why can't you simply use A's `wtxid` as the key in the `m_tx_results` map (even if A' is actually used)?
Why do you want the caller to try again? If the caller sees that the result type is `ResultType::WITNESS_SWAPPED` this should give them all the information they need?

Reply:
I agree with @t-bast, I think it should be the original wtxid with a witness swapped result type.

Reply:
I will open a PR to change this. Thanks very much for your reviews! :)

Reply:
Addressed in #23804